### PR TITLE
feat(data-warehouse): Select color of series on a query visualization

### PR DESCRIPTION
**Pull request by rossgray**
## Problem

At the moment it is not possible to change the color of a series on a query visualization.

### UX

Adds a color picker (the same one as for conditional formatting, thanks Tom!) to the series display options tab. See screen recording below.

https://github.com/user-attachments/assets/bae5eb29-8182-42ef-9925-16391ebdb92a

### Code

- I have have refactored the `ColorPickerButton` into its own file since it is now used in 2 different places (I also renamed it `ColourPickerButton` -> `ColorPickerButton` since we use the American spelling in most places and thought it best to be consistent).
- I also added a `colorChoices` prop so that for the series color we can pass in the set of colors currently used for series (this is different from the color palette used for conditional formatting)

### Limitations

- It is not ideal having to click off of the display settings before seeing the visualization update, however it is not straightforward to change this behaviour so decided to leave it for now.
- This is already an issue but there is not much space on the select for the series label
image
- In Metabase you can change the color of a series directly by clicking on the color glyph directly, without having to open up the settings tab (see below).
image
I thought about doing the same - I think it would require us to move the colour glyph out of the select to a separate button on the left (see below). The only thing is that there is then even less space to display the label, unless we change the layout a bit, so not sure whether to proceed with this or not?
image

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

I presume so.

## How did you test this code?

Local testing

### CODE REVIEW

File: frontend/src/queries/schema.ts [hunk 1]
```diff
@@ -668,6 +668,7 @@ export interface ChartSettingsFormatting {
 }
 
 export interface ChartSettingsDisplay {
+    color?: string
```
Comment (Gilbert09) [inline frontend/src/queries/schema.ts:671]: There should be a backend equivalent for this - generate it with `pnpm schema:build` command. Without it, I think saving an insight with a color will get rejected by the backend (I think)

**Review by Gilbert09 — APPROVED** [review]
Just the one change, then this is good - top stuff! 🚀

File: frontend/src/queries/schema.ts (hunk 1 above)
Comment (rossgray) [reply frontend/src/queries/schema.ts:671]: Ah right, I hadn't tested saving an insight. Will try what you said 👍

Comment (rossgray) [reply frontend/src/queries/schema.ts:673]: that should be fixed now

**Review by Gilbert09 — APPROVED** [review]

**Pull request merged**